src/libs/OptionsListUtils.ts

```diff
@@ -943,32 +943,32 @@ function getCategoryListSections(
         return categorySections;
     }
 
-    if (selectedOptions.length > 0) {
+    const selectedOptionNames = selectedOptions.map((selectedOption) => selectedOption.name);
+    const enabledAndSelectedCategories = sortedCategories.filter((category) => category.enabled || selectedOptionNames.includes(category.name));
+    const numberOfVisibleCategories = enabledAndSelectedCategories.length;
+
+    if (numberOfVisibleCategories < CONST.CATEGORY_LIST_THRESHOLD) {
         categorySections.push({
-            // "Selected" section
+            // "All" section when items amount less than the threshold
             title: '',
             shouldShow: false,
             indexOffset,
-            data: getCategoryOptionTree(selectedOptions, true),
+            data: getCategoryOptionTree(enabledAndSelectedCategories),
```

Comment:
When one subcategory is selected other subcategories of this category are indented without parent category name causing #37774